run/idp-sql/main.py

```diff
@@ -0,0 +1,304 @@
+            "DB_NAME": os.environ["DB_NAME"],
+            "DB_HOST": os.environ.get("DB_HOST", None),
+            "CLOUD_SQL_CONNECTION_NAME": os.environ["CLOUD_SQL_CONNECTION_NAME"],
+        }
+
+
+def init_tcp_connection_engine(
+    db_config: Dict[str, str]
+) -> sqlalchemy.engine.base.Engine:
+    creds = get_cred_config()
+    db_user = creds["DB_USER"]
+    db_pass = creds["DB_PASSWORD"]
+    db_name = creds["DB_NAME"]
+    db_host = creds["DB_HOST"]
+
+    # Extract host and port from db_host
+    host_args = db_host.split(":")
+    db_hostname, db_port = host_args[0], int(host_args[1])
+
+    pool = sqlalchemy.create_engine(
+        # Equivalent URL:
+        # postgres+pg8000://<db_user>:<db_pass>@<db_host>:<db_port>/<db_name>
+        sqlalchemy.engine.url.URL(
+            drivername="postgresql+pg8000",
+            username=db_user,  # e.g. "my-database-user"
+            password=db_pass,  # e.g. "my-database-password"
+            host=db_hostname,  # e.g. "127.0.0.1"
+            port=db_port,  # e.g. 5432
+            database=db_name,  # e.g. "my-database-name"
+        ),
+        **db_config,
+    )
+    pool.dialect.description_encoding = None
+    return pool
+
+
+# [START cloudrun_python_user_auth_sql_connect]
+def init_unix_connection_engine(
+    db_config: Dict[str, str]
+) -> sqlalchemy.engine.base.Engine:
+    creds = get_cred_config()
+    db_user = creds["DB_USER"]
+    db_pass = creds["DB_PASSWORD"]
+    db_name = creds["DB_NAME"]
+    db_socket_dir = creds.get("DB_SOCKET_DIR", "/cloudsql")
+    cloud_sql_connection_name = creds["CLOUD_SQL_CONNECTION_NAME"]
+
+    pool = sqlalchemy.create_engine(
+        # Equivalent URL:
+        # postgres+pg8000://<db_user>:<db_pass>@/<db_name>
+        #                         ?unix_sock=<socket_path>/<cloud_sql_instance_name>/.s.PGSQL.5432
+        sqlalchemy.engine.url.URL(
+            drivername="postgresql+pg8000",
+            username=db_user,  # e.g. "my-database-user"
+            password=db_pass,  # e.g. "my-database-password"
+            database=db_name,  # e.g. "my-database-name"
+            query={
+                "unix_sock": "{}/{}/.s.PGSQL.5432".format(
+                    db_socket_dir, cloud_sql_connection_name  # e.g. "/cloudsql"
+                )  # i.e "<PROJECT-NAME>:<INSTANCE-REGION>:<INSTANCE-NAME>"
+            },
+        ),
+        **db_config,
+    )
+    pool.dialect.description_encoding = None
+    return pool
+
+
+# [END cloudrun_python_user_auth_sql_connect]
+
+
+# This global variable is declared with a value of `None`, instead of calling
+# `init_connection_engine()` immediately, to simplify testing. In general, it
+# is safe to initialize your database connection pool when your script starts
+# -- there is no need to wait for the first request.
+db = None
+
+
+@app.before_first_request
+def create_tables() -> None:
+    global db
+    db = init_connection_engine()
+    # Create tables (if they don't already exist)s
```

Comment:
Can we update the votes table to pet_votes. From the node version, we found that there can be an issue between this sample and cloud sql sharing the same table name but not schema. 
```suggestion
    # Create pet_votes table if it doesn't already exist
```